packages/css-jss/src/createCss.js

```diff
@@ -0,0 +1,31 @@
+// @flow
+const createCss = (sheet, cache) => {
+  let ruleIndex = 0
+
+  function css() {
+    // eslint-disable-next-line prefer-rest-params
+    const args = arguments
+    const className = cache.get(args)
+
+    if (className) return className
+
+    const style = {}
+    let label = 'css'
+
+    for (const i in args) {
+      const arg = args[i]
+      if (arg) Object.assign(style, arg)
+    }
+    if ('label' in style) {
+      label = style.label
+      delete style.label
+    }
+    const key = `${label}${ruleIndex++}`
+    sheet.addRule(key, style)
```

Comment:
Will this deduplicate the same style created inside of component render?

Reply:
yes, if object refs are not the same, to avoid this, one will need to use a babel plugin (planned)

Reply:
Alternatively I could use JSON.string() and compare by that ...

Reply:
Does jss serialize styles internally? Maybe use serialized value to get the key?

Reply:
Nope, we optimized that away in JSS.

Reply:
Actually with this logic I can also remove the ref cache and just use string based keys and hope that user will use babel plugin to avoid the need for JSON.stringify()